tests/trx_generator/trx_generator.cpp

```diff
@@ -5,200 +5,294 @@
 
-      trx.set_reference_block(last_irr_block_id);
-      trx.expiration = fc::time_point::now() + trx_expiration;
-      trx.max_net_usage_words = 100;
       trx.sign(priv_key, chain_id);
       return signed_transaction_w_signer(trx, priv_key);
    }
 
-   vector<signed_transaction_w_signer> create_initial_transfer_transactions(const vector<action_pair_w_keys>& action_pairs_vector, uint64_t& nonce_prefix, uint64_t& nonce, const fc::microseconds& trx_expiration, const chain_id_type& chain_id, const block_id_type& last_irr_block_id) {
+   void transfer_trx_generator::create_initial_transfer_transactions(uint64_t& nonce_prefix, uint64_t& nonce) {
       std::vector<signed_transaction_w_signer> trxs;
-      trxs.reserve(2 * action_pairs_vector.size());
+      _trxs.reserve(2 * _action_pairs_vector.size());
 
-      for(action_pair_w_keys ap: action_pairs_vector) {
-         trxs.emplace_back(create_transfer_trx_w_signer(ap._first_act, ap._first_act_priv_key, nonce_prefix, nonce, trx_expiration, chain_id, last_irr_block_id));
-         trxs.emplace_back(create_transfer_trx_w_signer(ap._second_act, ap._second_act_priv_key, nonce_prefix, nonce, trx_expiration, chain_id, last_irr_block_id));
+      for (const action_pair_w_keys& ap : _action_pairs_vector) {
+         _trxs.push_back(create_trx_w_actions_and_signer({ap._first_act}, ap._first_act_priv_key, nonce_prefix, nonce, _config._trx_expiration_us, _config._chain_id, _config._last_irr_block_id));
+         _trxs.push_back(create_trx_w_actions_and_signer({ap._second_act}, ap._second_act_priv_key, nonce_prefix, nonce, _config._trx_expiration_us, _config._chain_id, _config._last_irr_block_id));
       }
-
-      return trxs;
    }
 
-   void update_resign_transaction(signed_transaction& trx, fc::crypto::private_key priv_key, uint64_t& nonce_prefix, uint64_t& nonce, const fc::microseconds& trx_expiration, const chain_id_type& chain_id, const block_id_type& last_irr_block_id) {
+   void trx_generator_base::update_resign_transaction(signed_transaction& trx, const fc::crypto::private_key& priv_key, uint64_t& nonce_prefix, uint64_t& nonce, const fc::microseconds& trx_expiration,
+                                                      const chain_id_type& chain_id, const block_id_type& last_irr_block_id) {
       trx.context_free_actions.clear();
-      trx.context_free_actions.emplace_back(action({}, config::null_account_name, name("nonce"), fc::raw::pack(std::to_string(nonce_prefix) + ":" + std::to_string(++nonce) + ":" + fc::time_point::now().time_since_epoch().count())));
-      trx.set_reference_block(last_irr_block_id);
-      trx.expiration = fc::time_point::now() + trx_expiration;
+      trx.context_free_actions.emplace_back(vector<permission_level>(), config::null_account_name, name("nonce"),
+         fc::raw::pack(std::to_string(_config._generator_id) + ":" + std::to_string(nonce_prefix) + ":" + std::to_string(++nonce) + ":" +
+         fc::time_point::now().time_since_epoch().count()));
```

Comment:
I think we'd have more consistency combining lines 55 and 56. It's a single argument to the function and the line would still be shorter than line 51 above. Since this is aesthetic and of no real consequence, please feel free to disregard and mark this comment resolved. :-)